core/metrics-spi/src/main/java/software/amazon/awssdk/metrics/SdkMetrics.java

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.metrics;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import software.amazon.awssdk.annotations.SdkProtectedApi;
+import software.amazon.awssdk.metrics.meter.Metric;
+
+/**
+ * A enum representing the metrics supported by the SDK. All metrics collected by the SDK will be declared in this
+ * enum class along with the metric's corresponding {@link MetricCategory}.
+ */
+// TODO List is not complete
+@SdkProtectedApi
+public enum SdkMetrics implements Metric {
+
+    /**
+     * Service ID of the AWS service that the API request is made against
+     */
+    Service("Service", MetricCategory.Default),
+
+    /**
+     * The name of the AWS operation the request is made to
+     */
+    Api("Api", MetricCategory.Default),
+
+    /**
+     * The total time taken to finish (success or fail) a request (inclusive of all retries)
+     */
+    ApiCallLatency("ApiCallLatency", MetricCategory.Default),
```

Comment:
Api -> Request?

Reply:
I was going with CSM naming convention.
ApiCall for main request
ApiCallAttempt for each attempt of the request